ui/src/app/pages/workspace/create-billing-account-modal.tsx

```diff
@@ -1,76 +1,410 @@
+  };
+
+  const sendCreateBillingEmail = fp.flow(
+    withErrorModal({
+      title: 'Failed To Send Email',
+      message: 'An error occurred trying to send email. Please try again.',
+    })
+  )(async() => {await profileApi().sendBillingSetupEmail(
+    {
+      phone: phoneNumber,
+      paymentMethod: useCreditCard ? BillingPaymentMethod.CREDITCARD : BillingPaymentMethod.PURCHASEORDER,
+      isNihFunded: nihFunded,
+      institution: verifiedInstitutionalAffiliation.institutionDisplayName
+    }
+  );
+    setCurrentStep(4);
+  });
 
-            <TextColumn>
-              <div>Add [ {this.props.profileState.profile.username} ] as a 'User' to your Google Cloud billing account.</div>
-              <a href='https://console.cloud.google.com/billing' target='_blank'>Add user</a>
-            </TextColumn>
+  return <Modal width={650} onRequestClose={() => onClose()}>
+        <ModalBody style={{marginTop: '-4'}}>
+          <FlexColumn>
+            <FlexRow style={{alignItems: 'center', width: '620px', marginBottom: '0.8rem'}}>
+              <GoogleCloudLogoSvg style={{height: '33px', width: '207px', marginLeft: '-0.5remx', marginRight: '0.5rem'}}/>
+              <div style={styles.line}></div>
+              <div style={{paddingTop: 5, marginLeft: '1rem', marginRight: '2rem'}}>
+                <div style={styles.textHeader}>Create billing account</div></div>
+                <div style={stylesFunction.stepButtonCircle(currentStep, 1)}>1</div>
+                <div style={stylesFunction.stepButtonCircle(currentStep, 2)}>2</div>
+                <div style={stylesFunction.stepButtonCircle(currentStep, 3)}>3</div>
+            </FlexRow>
+            {currentStep === 0 && <TextColumn>
+              <div style={styles.textNormal}>Billing accounts are managed via Google Cloud Platform™ service.</div>
+              <div style={styles.textNormal}>Learn more on how to set up a billing account.</div>
+            </TextColumn>}
+            {currentStep !== 0 && currentStep !== 4 && <TextColumn>
+              <div style={styles.textNormal}>Submit your information below to receive billing and additional information from
+                a Google billing partner representative.</div>
+            </TextColumn>}
           </FlexColumn>
-        </ModalFooter>
+        </ModalBody>
+        <hr style={{
+          width: '100%', backgroundColor: '#979797', borderWidth: '0px', height: '1px',
+          marginTop: '0.5rem',
+          marginBottom: '0.5rem'
+        }}/>
 
-        <FontAwesomeIcon
-          icon={faTimes}
-          size='lg'
-          style={{
-            color: colors.accent,
-            position: 'absolute',
-            top: '1rem',
-            right: '1rem',
-            cursor: 'pointer'
-          }}
-          onClick={() => this.props.onClose()}
-        />
-      </Modal>;
-    }
-  }
-);
+    {currentStep === 0 && <ModalFooter data-test-id='step-0-modal' style={{marginTop: 0, justifyContent: 'flex-start'}}>
```

Comment:
nit: consider switchCase() from our utils/index.ts, since these cases are all mutually exclusive

It also feels like the `ModalFooter` piece is common across all of these, and possibly the `flexrow` too - would pull out what common pieces you can outside of the switch

Reply:
Done with switchCase. But I fail to extract common ModalFooter  and flexrow because they are slightly different. 
Like 1st step no Back button, the others have, so I have to create one more layer of Flex for the other steps. 
And the final step is Submit with a different function. Extracting common method end up having if(step === 1), else if(step ===2)

Reply:
Sorry, yes - here is one similar `switchCase` example: https://github.com/all-of-us/workbench/blob/master/ui/src/app/pages/analysis/runtime-panel.tsx#L1063. This looks good.

I don't think you need a for loop for ModalFooter. The code looks identical, just parameterize it with the `currentStep`:

```
<ModalFooter data-test-id={`step-${currentStep}-modal`} style={{marginTop: 0, justifyContent: 'flex-start'}}>
  {switchCase(...
</ModalFooter>